rust/worker/src/execution/dispatcher.rs

```diff
@@ -97,7 +98,11 @@ impl Dispatcher {
         // If a worker is waiting for a task, send it to the worker in FIFO order
         // Otherwise, add it to the task queue
         match self.waiters.pop() {
-            Some(channel) => match channel.reply_to.send(task).await {
+            Some(channel) => match channel
+                .reply_to
+                .send(task, Some(Span::current().clone()))
```

Comment:
Yeah, I gave this some thought. I was running into an issue where the dispatcher would pass in a span to the worker and by the time the worker executed with this span as a child, the parent span gets GC'd because the RAII guard that the dispatcher creates goes out of scope. So, we need some ref counting at the very least here.

I thought of trying out `Arc` then but the problem with that is the genesis span that I instrument at the beginning of the query_vectors RPC in query_service is created automatically by tracing library (since I instrument it with the `#instrument` wrapper) so I can't create an `Arc` for that. Hence, cloning.